tests/codegen/waiter-tests/src/test/kotlin/com/test/WaiterTest.kt

```diff
@@ -5,90 +5,774 @@
+                    structs = listOf(
+                        Struct {
+                            primitives = EntityPrimitives { string = Enum.One.value }
+                            enums = listOf()
+                        },
+                    )
+                }
+            },
+            GetEntityResponse {
+                lists = EntityLists {
+                    structs = listOf(
+                        Struct {
+                            primitives = EntityPrimitives { string = Enum.One.value }
+                            enums = listOf(Enum.Two)
+                        },
+                    )
+                }
+            },
+            GetEntityResponse {
+                lists = EntityLists {
+                    structs = listOf(
+                        Struct {
+                            primitives = EntityPrimitives { string = Enum.One.value }
+                            enums = listOf(Enum.Two, Enum.One)
+                        },
+                    )
+                }
+            },
+        ),
+        WaitersTestClient::waitUntilHasStructWithStringInEnumList,
+    )
+    @Test fun testHasStructWithEnumInStringList() = successTest(
+        listOf(
+            GetEntityResponse { },
+            GetEntityResponse { lists = EntityLists { } },
+            GetEntityResponse {
+                lists = EntityLists { structs = listOf() }
+            },
+            GetEntityResponse {
+                lists = EntityLists { structs = listOf(Struct {}) }
+            },
+            GetEntityResponse {
+                lists = EntityLists {
+                    structs = listOf(
+                        Struct { enums = listOf() },
+                    )
+                }
+            },
+            GetEntityResponse {
+                lists = EntityLists {
+                    structs = listOf(
+                        Struct {
+                            primitives = EntityPrimitives { string = Enum.One.value }
+                            enums = listOf()
+                        },
+                    )
+                }
+            },
+            GetEntityResponse {
+                lists = EntityLists {
+                    structs = listOf(
+                        Struct {
+                            primitives = EntityPrimitives { enum = Enum.One }
+                            strings = listOf(Enum.Two.value)
+                        },
+                    )
+                }
+            },
+            GetEntityResponse {
+                lists = EntityLists {
+                    structs = listOf(
+                        Struct {
+                            primitives = EntityPrimitives { enum = Enum.One }
+                            strings = listOf(Enum.Two.value, Enum.One.value)
+                        },
+                    )
+                }
+            },
+        ),
+        WaitersTestClient::waitUntilHasStructWithEnumInStringList,
+    )
```

Comment:
**Question**: In general, this test suite is very comprehensive and clear so great job on that! These longer tests, however, are a little hard to read because of all the boilerplate. Since this is an integration test, did you consider using serialized responses instead of object literals? That might shorten this test representation:

```kotlin
@Test fun testHasStructWithEnumInStringList() = successTest(
    WaitersTestClient::waitUntilHasStructWithEnumInStringList,
    """{}""",
    """{ "lists": {} }""",
    """{ "lists": { "structs": [] } }""",
    """{ "lists": { "structs": [ {} ] } }""",
    """{ "lists": { "structs": [ { "enums": [] } ] } }""",
    """{ "lists": { "structs": [ { "primitives": { "string": "ONE" }, "enums": [] } ] } }""",
    """{ "lists": { "structs": [ { "primitives": { "enum": "ONE" }, "strings": [ "TWO" ] } ] } }""",
    """{ "lists": { "structs": [ { "primitives": { "enum": "ONE" }, "strings": [ "TWO", "ONE" ] } ] } }""",
)
```

Reply:
Though it _IS_ an integration test, there's still a main topic of validation within (waiters), I don't think we should implicitly bring in serde concerns into these tests.